cloudinstall/juju.py

```diff
@@ -40,6 +40,31 @@ def __init__(self, juju):
         self._juju_status = None
         self.valid_states = ['pending', 'started', 'down']
 
+    def get_agents_states(self):
+        """ Returns list of deployed services and their agent-state """
+        states = []
+        for svc in self.services:
+            for unit in svc.units:
+                states.append((svc.service_name, unit.agent_state))
+        return states
+
+    def all_agents_started(self):
+        """ Check status of all deployed agents
+
+        :param list svcs: List of services to check or empty for calling
+                          service
+        :rtype: :class:`~cloudinstall.service.Unit`
+        :returns: True if all svcs are started, False otherwise
+        """
+        status_res = []
+        for svc, state in self.get_agents_states():
+            if state == "started":
+                status_res.append(True)
+                continue
+            else:
+                return False
+        return all(status_res)
```

Comment:
you don't need the `continue` in this loop.

A shorter and I think slightly more pythonic way to do this function is here:

```python
return all([state == "started" for _, state in self.get_agents_states()])
```